cmd/gitannex/gitannex.md

```diff
@@ -0,0 +1,38 @@
+Rclone's gitannex subcommand enables git-annex to store and retrieve content
+from an rclone remote. It expects to be run by git-annex, not directly by users.
+It is an "external special remote program" as defined by git-annex.
+
+Installation on Linux
+---------------------
+
+1. Create a symlink and ensure it's on your PATH. For example:
+
+        ln -s "$(realpath rclone)" "$HOME/bin/git-annex-remote-rclone-builtin"
```

Comment:
I need to learn this dark magic (if it works ;-) yet to get a chance to try, but I have concerns ).

But ideally IMHO `rclone` itself should just install a shim itself in the PATH. The trickery that it would be different for *nix (shell script is enough) and Windows (needs .bat or .exe?)

Reply:
Please do give it a try! I'm happy to make changes if there's something broken or risky.

The dark magic happens in gitannex.go's `maybeTransformArgs()`, which is called by `init()`. Basically, if `argv[0]` looks like "git-annex-remote-rclone-builtin", we insert the `gitannex` subcommand in the second position. There's actually already precedent in rclone for rewriting arguments like this: <https://rclone.org/commands/rclone_mount/#rclone-as-unix-mount-helper>.

It's decidedly a hack, but it means we don't have to write any platform-specific wrapper scripts. I think we can do the rough equivalent of `ln -s` on Windows with `mklink`, but I don't have a Windows machine handy to try it out. 

I see a few paths forward that enable us to remove the symlink step from the setup instructions:

* If git-annex adds a special case for this remote, i.e. @joeyh resolves this [todo](https://git-annex.branchable.com/todo/external_special_remotes_not_using_git-annex-remote_in_name/), users could just set the `type` or `externaltype` to whatever git-annex expects. No need for a symlink.
* We could write wrapper scripts for every supported platform. Honestly, this is probably not that hard, but I'm generally wary of subtleties introduced by shell scripts.
* We could write wrapper scripts for every supported platform and add a sub-subcommand (something like `rclone gitannex installwrapper`) that installs the platform-specific wrapper somewhere. That location could be alongside the `rclone` binary, but that makes me nervous. It could be a user-selected bin directory, but then the user still has to manage their PATH.
* We could write wrapper scripts for every supported platform and ship them in rclone releases OR we could even ship relative symlinks in the releases, I suppose. This solves the package manager issue, but I'm not sure how project maintainers would feel about adding files to the release.

Reply:
sorry -- my bad -- I was just too tired I guess. Instead of `realpath rclone` I saw `rclone gitannex` ;-)

Thanks for the analysis!  yeah - we need @ncw's input on this.  I guess such a wrapper could be done directly in `go` and just come with `rclone` package "natively" to not mess with "platform specific solutions" (no symlinks on windows really AFAIK).

Reply:
> I see a few paths forward that enable us to remove the symlink step from the setup instructions:

You could get rclone to create the symlink on the first writable element in PATH it finds...

I'd avoid scripting if possible - Go is marvelously consistent between platforms whereas shell/powershell isn't!

If we can build the logic into rclone then that makes it a one binary deploy which is what we aim for.